src/cartridge.c

```diff
@@ -0,0 +1,542 @@
+#include "cartridge.h"
+#include "files.h"
+
+#include <ctype.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+struct x16cartridge_header Cartridge_info;
+uint8_t *CART = NULL;
+char *Cartridge_path = NULL;
+char *Cartridge_nvram_path = NULL;
+
+const char Cartridge_magic_number[CART_MAGIC_NUMBER_SIZE] = { 'C', 'X', '1', '6', ' ', 'C', 'A', 'R', 'T', 'R', 'I', 'D', 'G', 'E', '\r', '\n' };
+const char Cartridge_current_version[CART_VERSION_SIZE]   = { '0', '1', '.', '0', '0', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '};
+
+static void 
+initialize_cart_bank(uint8_t *mem, bool randomize)
+{
+	if(randomize) {
+		for(int i=0; i<CART_BANK_SIZE; ++i) {
+			mem[i] = rand();
+		}
+	} else {
+		memset(mem, 0, CART_BANK_SIZE);
+	}
+}
+
+bool 
+cartridge_load(const char *path, bool randomize)
+{
+	const char *extension = file_find_extension(path, NULL);
+	if(strncmp(extension, ".crt", 4) != 0) {
```

Comment:
This case-sensitive comparison might be a cause of minor annoyance for people on case-preserving case-insensitive filesystems.  Just a small consideration.  I realize now this is probably also true for the compressed file extensions.

Reply:
Worthwhile. Too bad strnicmp is a non-standard C function. Added a similar implementation that should provide a case-insensitive compare.